python/ray/train/examples/mosaic_cifar10_example.py

```diff
@@ -31,8 +30,8 @@ def trainer_init_per_worker(config):
         [transforms.ToTensor(), transforms.Normalize(mean, std)]
     )
 
-    data_directory = tempfile.mkdtemp(prefix="cifar_data")
-    with FileLock(os.path.join(data_directory, "data.lock")):
+    data_directory = os.path.expanduser("~/data")
+    with FileLock(os.path.expanduser("~/data.lock")):
```

Comment:
Can you help me understand the change here? 

Previously, each worker locks on different file, so they all download the datasets in parallel. Now one worker download the datasets once, then when creating `datasets.CIFAR10(...)`, the following workers will not download the datasets again? 

Is it be cause the network igress bandwidth is too small, and N workers download at the same time will takes longer time than only one worker download the dataset?

Reply:
i think the problem was that there would be multiple tests that download the data 4 times, and if you just keep it at a fixed ~/data, then every test after the 1st one will just use the already downloaded directory

I basically messed up this logic in this previous PR: https://github.com/ray-project/ray/commit/78d6ef18959b2ac042d330a4e30ddebef018804a#diff-00158017bc9d4ef2293d120c0a64970bb3b9aef546fff57f0559cb8f5a8d1544L32-L33